.github/workflows/snyk.yml

```
@@ -0,0 +1,50 @@
+# GitHub action to run Snyk check for vulnerabilities in all requirement files.
+
+name: run-snyk
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  snyk:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.10"]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        id: setup_python
+        with:
+          python-version: ${{ matrix.python-version}}
```

```suggestion
    steps:
      - uses: actions/checkout@v3

      - name: Set up Python
        uses: actions/setup-python@v4
        with:
          python-version: "3.10"
```

simplifying a bit more here. No need for the test grid.